libs/components/forms/src/lib/modules/checkbox/checkbox.component.ts

```diff
@@ -162,55 +173,63 @@ export class SkyCheckboxComponent implements ControlValueAccessor, OnInit {
    * @default false
    */
   @Input()
-  set required(value: boolean) {
-    this._required = SkyFormsUtility.coerceBooleanProperty(value);
+  set required(value: boolean | undefined) {
+    this.#_required = SkyFormsUtility.coerceBooleanProperty(value);
   }
 
   get required(): boolean {
-    return this._required;
+    return this.#_required;
   }
 
   /**
    * Fires when users select or deselect the checkbox.
    */
   @Output()
   public get checkedChange(): Observable<boolean> {
-    return this._checkedChange;
+    return this.#_checkedChange;
   }
 
   /**
    * Fires when the selected value changes.
    */
   @Output()
   public get disabledChange(): Observable<boolean> {
-    return this._disabledChange;
+    return this.#_disabledChange;
   }
 
-  private isFirstChange = true;
+  #isFirstChange = true;
+
+  #_checked = false;
 
-  private _checkboxType: string;
+  #_checkedChange = new BehaviorSubject<boolean>(this.#_checked);
 
-  private _checked = false;
+  #_checkboxType = 'info';
 
-  private _checkedChange = new BehaviorSubject<boolean>(this._checked);
+  #_disabled = false;
 
-  private _disabled = false;
+  #_disabledChange = new BehaviorSubject<boolean>(this.#_disabled);
 
-  private _disabledChange = new BehaviorSubject<boolean>(this._disabled);
+  #_id: string;
 
-  private _required = false;
+  #_required = false;
 
-  constructor(@Self() @Optional() private ngControl: NgControl) {
-    if (this.ngControl) {
-      this.ngControl.valueAccessor = this;
+  #ngControl: NgControl | undefined;
+
+  constructor(@Self() @Optional() ngControl: NgControl) {
+    this.#ngControl = ngControl;
+    if (ngControl) {
+      ngControl.valueAccessor = this;
     }
+
+    this.#_id = `sky-checkbox-${++nextId}`;
```

Comment:
I'd find a way to avoid referencing any `#_` properties anywhere other than the `get` and `set` accessors. Following the guidance in the comment about only calling `++nextId` once might cause you to refactor code and fix this issue.

Reply:
Done